drivers/docker/config.go

```diff
@@ -206,6 +206,18 @@ var (
 		// extra docker labels, globs supported
 		"extra_labels": hclspec.NewAttr("extra_labels", "list(string)", false),
 
+		// logging options
+		"logging": hclspec.NewDefault(hclspec.NewBlock("logging", false, hclspec.NewObject(map[string]*hclspec.Spec{
+			"type":   hclspec.NewAttr("type", "string", false),
+			"config": hclspec.NewBlockAttrs("config", "string", false),
+		})), hclspec.NewLiteral(`{
+			type = "json-file" 
+			config = {
+				max-file = "2"
```

Comment:
Especially after reading the docs updates, I'm realizing there's a bit of a conceptual gap here we need to solve. The log driver configuration with `max-file` and `max-size` is not the same as the Nomad [`logs`](https://www.nomadproject.io/docs/job-specification/logs) block; one is Docker's own logs and the other is the Nomad allocation logs.

I feel like the original issue #6563 that @powellchristoph opened was ambiguous as to whether it was understood that these are different values. If so, we should probably clarify that further in the docs. I'm also wondering whether there's value in being able to set the Nomad defaults in the client config (either as well or instead of).

Reply:
ha, I never knew that nomad has a `logs` block. For me the main goal is to send the task logs somewhere else. For docker the easiest way to do this is via a logdriver. I think in an ideal world, nomad would be able to send those logs away (with metadata attached).

As for the `logs` block, no idea -- I'll leave it up to you to decide what to do with it. Personally I think it would belong in another PR to keep to scope small.

Reply:
> I think in an ideal world nomad would be able to send those logs away (with metadata attached) -- for all task drivers without relying on some hacks like docker logging drivers.

Agreed. We have an issue open for some kind of log driver plugin concept (https://github.com/hashicorp/nomad/issues/9211) but that hasn't quite made its way onto the near-term roadmap yet.

> As for the logs block, no idea -- I'll leave it up to you to decide what to do with it. Personally I think it would belong in another PR to keep to scope small.

I'm inclined to agree, but lemme run that by a couple of other folks on the team just to get a little consensus.

Reply:
Ok, had a quick discussion about that internally and I think we're just going to keep the code here as is. I'm going to push a commit onto this PR just to add a few choice warnings about the difference between the two configuration values and then we'll get this merged.

Reply:
Actually, I'm happy with what you've got here. Let's merge this.